### PR TITLE
THD Attributes register db/examined_row_count/row_count_func/sent_row_count

### PR DESCRIPTION
**Pull request by hongdongjian**
THD Attributes register m_db/m_examined_row_count/m_row_count_func/m_sent_row_count

### CODE REVIEW

File: sql/server_component/mysql_thd_attributes_imp.cc [hunk 1]
```diff
@@ -50,6 +50,14 @@ DEFINE_BOOL_METHOD(mysql_thd_attributes_imp::get,
         *((bool *)inout_pvalue) = t->is_server_upgrade_thread();
       } else if (!strcmp(name, "is_init_file_thread")) {
         *((bool *)inout_pvalue) = t->is_init_file_system_thread();
+      } else if (!strcmp(name, "m_db")) {
+        *((LEX_CSTRING *)inout_pvalue) = t->db();
```
Comment (gkodinov) [inline sql/server_component/mysql_thd_attributes_imp.cc]: it doesn't work this way: LEX_CSTRING doesn't for example carry charset info. You're better off using mysql_string strings (this is the official charset compatible way to return data) or you need to convert to a known charset (utf-8) and copy it in a buffer.

File: sql/server_component/mysql_thd_attributes_imp.cc [hunk 2]
```diff
@@ -50,6 +50,14 @@ DEFINE_BOOL_METHOD(mysql_thd_attributes_imp::get,
...
+      } else if (!strcmp(name, "m_examined_row_count")) {
+        *((ha_rows *)inout_pvalue) = t->get_examined_row_count();
```
Comment (gkodinov) [inline sql/server_component/mysql_thd_attributes_imp.cc]: ha_rows is an internal type. Please use one of the C types that's closest. e.g. an unsigned long long.

File: sql/server_component/mysql_thd_attributes_imp.cc [hunk 3]
```diff
@@ -50,6 +50,14 @@ DEFINE_BOOL_METHOD(mysql_thd_attributes_imp::get,
         *((bool *)inout_pvalue) = t->is_server_upgrade_thread();
       } else if (!strcmp(name, "is_init_file_thread")) {
         *((bool *)inout_pvalue) = t->is_init_file_system_thread();
+      } else if (!strcmp(name, "m_db")) {
+        *((LEX_CSTRING *)inout_pvalue) = t->db();
+      } else if (!strcmp(name, "m_examined_row_count")) {
```
Comment (gkodinov) [inline sql/server_component/mysql_thd_attributes_imp.cc]: don't to the "m_" prefix in the names please.

**Review by gkodinov — CHANGES_REQUESTED** [review]
I'd also add (empty) switch cases in set to signify that these are taken but aren't settable. 
And I'd consider some mtr or unit test.

**Comment by hongdongjian** [comment]
I confirm the code being submitted is offered under the terms of the OCA, and that I am authorized to contribute it.

**Comment by hongdongjian** [comment]
> I'd also add (empty) switch cases in set to signify that these are taken but aren't settable. And I'd consider some mtr or unit test.

Thank you for your guidance. I have made the corresponding changes according to your advice. When you have time, could you please help me take a look and I look forward to your reply.

**Pull request closed without merging**